packages/react-app/components/notifications/OnboardingBatch.jsx

```diff
@@ -1,38 +1,133 @@
+      number: NEXT_BATCH,
+      status: "candidate",
+    };
+
+    let signMessage;
+    try {
+      signMessage = await getUpdateBatchSignMessage(address, batchData);
+    } catch (error) {
+      toast({
+        description: "Sorry, the server is overloaded. 🧯🚒🔥",
+        status: "error",
+        variant: toastVariant,
+      });
+      telegramWindow.close();
+      return;
+    }
+
+    let signature;
+    try {
+      signature = await userProvider.send("personal_sign", [signMessage, address]);
+    } catch (error) {
+      toast({
+        description: "Couldn't get a signature from the Wallet",
+        status: "error",
+        variant: toastVariant,
+      });
+      telegramWindow.close();
+      return;
+    }
+
+    try {
+      await postUpdateBatch(address, signature, batchData);
+      toast({
+        description: "Your batch number has been updated",
+        status: "success",
+        variant: toastVariant,
+      });
+
+      if (typeof onUpdate === "function") {
+        onUpdate();
+      }
+
+      telegramWindow.location.href = NEXT_BATCH_TELEGRAM;
+    } catch (error) {
+      toast({
+        status: "error",
+        description: "Can't save your batch number. Please try again.",
+        variant: toastVariant,
+      });
+      telegramWindow.close();
+    }
+  };
 
-const OnboardingBatch = ({ notification, onMarkAsRead }) => {
   return (
     <Flex background="linear-gradient(180deg, #EAFFA9 24.48%, #EDEFFF 100%)" position="relative">
       <Flex width={{ base: "100%", md: "60%" }} px="30px" py="30px" direction="column">
         <Box maxW="200px">
-          <Image src="/assets/onboarding_batches_logo.png" alt="onboarding batches illustarion: futuristic train" wi />
+          <Image src="/assets/onboarding_batches_logo.png" alt="onboarding batches illustration: futuristic train" />
         </Box>
         <Text fontSize="30px" fontWeight="bold" lineHeight="1.2" color="gray.800">
           Welcome to BuidlGuidl!
         </Text>
         <Text fontSize="sm" my="2" color="gray.800">
-          To help you get started in the Ethereum ecosystem, we've created the BuidlGuidl onboarding batch program. Dive into end-to-end
-          dApp development, receive mentorship from BG members, and learn how to collaborate with fellow developers in
-          open-source projects.
+          To help you get started in the Ethereum ecosystem, we've created the BuidlGuidl onboarding batch program. Dive
+          into end-to-end dApp development, receive mentorship from BG members, and learn how to collaborate with fellow
+          developers in open-source projects.
         </Text>
         <Spacer />
-        <Flex alignItems="center">
-          <Link href={NEXT_BATCH_TELEGRAM} isExternal>
-            <Button colorScheme="blue" mr="2" size="sm">
-              Join Telegram
-            </Button>
-          </Link>
+        <Flex alignItems="center" flexDirection="column">
+          {!hasTelegram ? (
```

Comment:
Question: Why do we force people to have their TG set up? Is it just to make it easier for admins?

(Just thinking out loud and trying to have the less possible barriers)